src/components/Giscus/Giscus.js

```diff
@@ -0,0 +1,53 @@
+import Head from '@docusaurus/Head';
+import { useLocation } from '@docusaurus/router';
+import { useColorMode } from '@docusaurus/theme-common';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import Giscus from "@giscus/react";
+import React from 'react';
+import styles from './styles.module.css';
+
+/**
+ * Adds a meta-tag to ensure new dicussion always add the canonical link in the initial post.
+ * Avoids any localhost etc. if first reaction/post was made from a preview build
+ *
+ * See https://github.com/giscus/giscus/blob/main/ADVANCED-USAGE.md#giscusbacklink
+ */
+export function GiscusHead() {
+  const {
+    siteConfig: { url: siteUrl },
+  } = useDocusaurusContext();
+  const { pathname } = useLocation();
+  const canonicalUrl = siteUrl + useBaseUrl(pathname);
+
+  return (
+    <Head>
+      <meta name="giscus:backlink" content={canonicalUrl} />
```

Comment:
Always add version here? Possible to get dynamically without hardcoding /docs/ = docs/v5 for stable like we do in _redirects